integration-tests/api/__tests__/admin/analytics-config.js

```diff
@@ -0,0 +1,215 @@
+      // Create config
+      await simpleAnalyticsConfigFactory(dbConnection)
+
+      const api = useApi()
+      const response = await api.get(`/admin/analytics-configs`, adminReqConfig)
+
+      expect(response.data).toMatchSnapshot({
+        analytics_config: {
+          id: expect.any(String),
+          user_id: "admin_user",
+          opt_out: false,
+          anonymize: false,
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+      })
+    })
+
+    it("should return 404 if no config exists", async () => {
+      const api = useApi()
+      try {
+        await api.get(`/admin/analytics-configs`, adminReqConfig)
+      } catch (e) {
+        expect(e.response.status).toEqual(404)
+        expect(e.response.data.message).toEqual(
+          "No analytics config found for user with id: admin_user"
+        )
+      }
+    })
+  })
+
+  describe("POST /admin/analytics-config", () => {
+    beforeEach(async () => {
+      try {
+        await adminSeeder(dbConnection)
+      } catch (e) {
+        console.error(e)
+      }
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      await db.teardown()
+    })
+
+    it("should create a new config for logged in user", async () => {
+      const api = useApi()
+      const response = await api
+        .post(
+          `/admin/analytics-configs`,
+          {
+            opt_out: true,
+            anonymize: true,
+          },
+          adminReqConfig
+        )
+        .catch((e) => console.log(e))
+
+      expect(response.data).toMatchSnapshot({
+        analytics_config: {
+          id: expect.any(String),
+          user_id: "admin_user",
+          opt_out: true,
+          anonymize: true,
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+      })
+    })
+  })
+
+  describe("POST /admin/analytics-config/update", () => {
+    beforeEach(async () => {
+      try {
+        await adminSeeder(dbConnection)
+      } catch (e) {
+        console.error(e)
+      }
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      await db.teardown()
+    })
+
+    it("should update the config of the logged in user", async () => {
+      // Create config
+      await simpleAnalyticsConfigFactory(dbConnection)
+
+      const api = useApi()
+      const response = await api.post(
+        `/admin/analytics-configs/update`,
+        {
+          opt_out: true,
+        },
+        adminReqConfig
+      )
+
+      expect(response.data).toMatchSnapshot({
+        analytics_config: {
+          id: expect.any(String),
+          user_id: "admin_user",
+          opt_out: true,
+          anonymize: false,
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+      })
+    })
+
+    it("should fail update the config of the logged in user if a config does not exist", async () => {
+      const api = useApi()
```

Comment:
add expect assertions

Reply:
suggestion: We should use .catch instead of try catch block in order to always reach the expect. and therefore the expect assertions become redundant